src/drivers/driver.ts

```diff
@@ -614,10 +620,114 @@ export abstract class CMakeDriver implements vscode.Disposable {
     await this.doPreCleanConfigure();
     this.configRunning = false;
 
-    return this.configure(extra_args, consumer);
+    return this.configure(trigger, extra_args, consumer);
+  }
+
+  async testCompilerVersion(program: string, cwd: string, arg: string | undefined, regexp: RegExp): Promise<string> {
+    const args = [];
+    if (arg) {
+      args.push(arg);
+    }
+    const child = this.executeCommand(program, args, undefined, {silent: true, cwd});
+    try {
+      const result = await child.result;
+      console.log(localize('command.version.test.return.code', 'Command version test return code {0}', nullableValueToString(result.retc)));
+      // cl.exe version information is printed on stderr when no source files are given
+      const versionLine = (program === "cl") ? result.stderr : result.stdout; //.split("\\n")[0];
+      const match = regexp.exec(versionLine);
+      return match ? match[2] : "error";
```

Comment:
how do we know the array has two elements?

Reply:
From the regular expressions constructed in getCompilerVersion. But indeed, there needs to be a way to be verified or enforced.

Reply:
And I'll remove that commented code in this snippet.

Reply:
You could add a field to your compilerAllowList entries that specifies which capture group the version string is in.